### PR TITLE
Provide some info on what actually happens

### PR DESCRIPTION
**Pull request by dirkx**
Provide some info on what actually happens and include CMS validation of the trust list signature (with a hardcoded link to the National PKI root of the Kingdom of the Netherlands.

### CODE REVIEW

**Comment by panzi** [comment]
Cool! But I just did a change that conflicts with this. Can you resolve those conflicts, squash the commits and also un-break pyzbar support? Maybe move `from pyzbar.pyzbar import decode as decode_qrcode # type: ignore` after `if args.image:` if you want to make loading that library optional. Thank you! :smile:

**Comment by dirkx** [comment]
Not quite proper across test/prod

File: verify_ehc.py [hunk 1]
```diff
@@ -794,18 +804,35 @@ def download_fr_certs(token: Optional[str] = None) -> CertList:
...
 def download_nl_certs(token: Optional[str] = None) -> CertList:
+    # Fetch the root certificate for the Netherlands; used to secure the
+    # trust list. Non fatal error if this fails.
+    response = requests.get(ROOT_URL_NL, headers={'User-Agent': USER_AGENT})
+    cacert = None
+    try:
+        if response.status_code == 200:
+          cacert = load_der_x509_certificate(response.content)
+    except:
+        pass
```
Comment (panzi) [inline verify_ehc.py:819]: `except: pass` is not good, error should be handled/logged, see `download_de_certs()`.

File: verify_ehc.py [hunk 2]
```diff
@@ -817,13 +820,22 @@ def download_nl_certs(token: Optional[str] = None) -> CertList:
...
     # The certificate chain in this pkcs#7 signature rolls up to the
     # rootkey of the Kingdom of the Netherlands (https://www.pkioverheid.nl)
     #
-    signature = b64decode(certs_json['signature'])
+    signature = certs_json['signature']
+
+    if cacert:
+        if pkcs7_detached.verify_detached_signature(payload, signature, cacert) != True:
+            print_err("Signature on the trustfile of the Netherlands did not verify against the countries National PKI root")
+            exit(1)
```
Comment (panzi) [inline verify_ehc.py:837]: Don't call `exit(1)`, `raise` an exception.

Comment (panzi) [reply verify_ehc.py:837]: Also are you sure this cannot be done with the `cryptography` library instead of yet another crypto library?

**Comment by dirkx** [comment]
Turned it into a warning. And quite intentionally not too fatal - as this relies on infrastructure that is not up 24x7.

So a fail here should not be in the critical path (that particular NL root certificate is already in the trust stores of the browsers, the operating system, openssl, etc by default).

**Comment by dirkx** [comment]
Yup - very sure.

**Comment by panzi** [comment]
I (rebased and) merged it with some changes. Thank you!

**Pull request closed without merging**